validate.go

```diff
@@ -0,0 +1,63 @@
+package cid
+
+import (
+	"fmt"
+
+	mh "github.com/multiformats/go-multihash"
+)
+
+var ErrPossiblyInsecureHashFunction = fmt.Errorf("potentially insecure hash functions not allowed")
+var ErrBelowMinimumHashLength = fmt.Errorf("hashes must be at least bytes long")
+
+const minimumHashLength = 20
+
+var goodset = map[uint64]bool{
+	mh.SHA2_256:     true,
+	mh.SHA2_512:     true,
+	mh.SHA3_224:     true,
+	mh.SHA3_256:     true,
+	mh.SHA3_384:     true,
+	mh.SHA3_512:     true,
+	mh.SHAKE_256:    true,
+	mh.DBL_SHA2_256: true,
+	mh.KECCAK_224:   true,
+	mh.KECCAK_256:   true,
+	mh.KECCAK_384:   true,
+	mh.KECCAK_512:   true,
+	mh.ID:           true,
+
+	// insecure
+	mh.SHA1:    false,
```

Comment:
This will, of course, break the git plugin.

Reply:
mmm, youre right.

Reply:
As suggested by @magik6k we should probably set SHA1 as secure, but add the check described here: https://github.com/multiformats/go-multihash/issues/57 to detect insecure hashes.

Reply:
@whyrusleeping this would still break the git plugin, but only in the very rare case when the content so happens to hash to one of the insure hashes, or am I missing something?

Reply:
yep

Reply:
@whyrusleeping yep to what?  Am I missing something or will this still break the git plugin in the very rare case.

Reply:
yes. This will still break the git plugin in that case.

Reply:
From IRC (slightly edits for clarity):

>**whyrusleeping**: juan wants this to make it into the next release
>*kevina*: whyrusleeping: I vote we allow "sha1" for now to avoid breaking the git plugin until we think of a better solution
>**whyrusleeping**: yeah, maybe.
>**whyrusleeping**: I think most of the concern we want to fix here is preimage attacks
>**whyrusleeping**: and theres no preimage attack for sha1
>**whyrusleeping**: just a collision
>**kevina** whyrusleeping, stebalien: so let's allow it with a note that it will eventually be marked insure once we implement something like /ipfs-insecure/
```